mllib/src/main/scala/org/apache/spark/ml/classification/LogisticRegression.scala

```diff
@@ -802,7 +840,11 @@ class LogisticRegressionModel private[spark] (
    * The behavior of this can be adjusted using [[thresholds]].
    */
   override protected def predict(features: Vector): Double = if (isMultinomial) {
-    super.predict(features)
+    if (isDefined(thresholds)) {
+      probability2prediction(scores(features))
```

Comment:
Actually, I read the code wrong originally. It was correct before. Still, we need tests so I am going add those.